File: Solution1/agadadotnet/Models/Shop.cs
using System;
using System.Collections.Generic;

namespace agadadotnet.Models;

public partial class Shop
{
    public int ShId { get; set; }

    public string Shname { get; set; } = null!;

    public string? RegNo { get; set; }

    public string? LicenseNo { get; set; }

    public int? UId { get; set; }

    public int? AId { get; set; }

    public string? Address { get; set; }

    public virtual Area? AIdNavigation { get; set; }

    public virtual User? UIdNavigation { get; set; }
}

File: entities/Shop.java
package com.example.demo.entities;

import jakarta.persistence.*;

@Entity
@Table(name = "shop")
public class Shop {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int shId;
    
    @Column(name = "shname", nullable = false)
    private String shname;
    
    @Column(name = "reg_no")
    private String regNo;
    
    @Column(name = "license_no")
    private String licenseNo;
    
    @ManyToOne
    @JoinColumn(name = "u_id")
    private User user;
    
    @ManyToOne
    @JoinColumn(name = "a_id")
    private Area area;
    
    @Column(name = "address")
    private String address;

    // Getters and Setters
}

